src/models/IConstructor.ts

```diff
@@ -1,7 +1,5 @@
 // Interface style:
-export default interface IConstructor<T> extends Function {
```

Comment:
you can't `export default` an interface yet (https://github.com/microsoft/TypeScript/issues/3792)

`ConstructorUnion` isn't used and its the same as your `IConstructor`